server/src/main/java/org/apache/iotdb/db/engine/compaction/utils/CompactionUtils.java

```diff
@@ -431,7 +431,10 @@ public static void merge(
       writer.endFile();
       targetResource.close();
     } finally {
-      writer.close();
+      if (writer.canWrite()) {
+        // avoid double close
```

Comment:
I feel we should set "canWrite" to volatile.
Because if two threads close concurrency， this will be not find.
This mistake is inevitable.
do you think ? @qiaojialin